app/views/shared/_footer.html.erb

```diff
@@ -20,7 +20,7 @@
   </script>
 </footer>
 <div class="container">
-  <% body_content = render 'shared/barcode_scanner' %>
+  <% body_content = render 'shared/barcode_scanner', product: @product %>
```

Comment:
If we add the footer partial to the homepage (as we planned to), will this code break the web? Since we don't have @product instance in the home page.

Reply:
It may well, but we will handle that problem when we get to it.